### PR TITLE
Show refunct-tas HUD at all times

### LINKED ISSUE #161: Fix refunct-tas HUD disappearing in Refunct menu
When you go into Refunct's menu, refunct-tas's HUD disappears.
To combat this, we're going to force the show hud bool to `true`

### PR DESCRIPTION
**Pull request by LukeSaward1**
Resolves #161.

### CODE REVIEW

File: rtil/src/native/character.rs [hunk 1]
```diff
@@ -82,6 +83,11 @@ impl AMyCharacter {
     pub fn set_rotation(&mut self, pitch: f32, yaw: f32, roll: f32) {
         self.controller_mut().rotation = FRotator { pitch, yaw, roll };
     }
+    pub fn show_hud() {
+        if !AMYHUD.is_none() {
+            unsafe { (*(*AMYHUD.get() as *mut AMyHud)).b_showhud.set_bit(1, true); }
+        }
+    }
```
Comment (oberien) [inline rtil/src/native/character.rs:90]: This function should be in `AMyHud`. Then you also don't need to make `AMYHUD` public (which it shouldn't be).

File: rtil/src/native/character.rs [hunk 2]
```diff
@@ -82,6 +83,11 @@ impl AMyCharacter {
     pub fn set_rotation(&mut self, pitch: f32, yaw: f32, roll: f32) {
         self.controller_mut().rotation = FRotator { pitch, yaw, roll };
     }
+    pub fn show_hud() {
+        if !AMYHUD.is_none() {
```
Comment (oberien) [inline rtil/src/native/character.rs:87]: Why is this check needed? `AMYHUD` should already be initialized before any rebo code is executed.

File: rtil/src/native/character.rs [hunk 3]
```diff
@@ -143,7 +149,7 @@ struct UCharacterMovementComponent {
 }
 
 #[repr(C)]
-struct APlayerController {
+pub struct APlayerController {
```
Comment (oberien) [inline rtil/src/native/character.rs:152]: unnecessary change

File: rtil/src/native/hud.rs [hunk 4]
```diff
@@ -8,12 +8,17 @@ use once_cell::sync::Lazy;
 
 use crate::native::ue::{FLinearColor, FString, FVector};
 use crate::native::{AHUD_DRAWLINE, AHUD_DRAWTEXT, AHUD_PROJECT, AHUD_GETTEXTSIZE};
+use crate::native::character::APlayerController;
```
Comment (oberien) [inline rtil/src/native/hud.rs:11]: unnecessary import

File: rtil/src/native/hud.rs [hunk 5]
```diff
@@ -8,12 +8,17 @@ use once_cell::sync::Lazy;
...
 use crate::threads::ue;
 use crate::statics::Static;
 
-static AMYHUD: Lazy<Static<usize>> = Lazy::new(Static::new);
+pub static AMYHUD: Lazy<Static<usize>> = Lazy::new(Static::new);
```
Comment (oberien) [inline rtil/src/native/hud.rs:15]: unnecessary `pub` given you move the `show_hud` function into `AMyHud`

File: rtil/src/native/mod.rs [hunk 6]
```diff
@@ -32,6 +32,11 @@ mod level_state;
 mod gameinstance;
 mod platform_misc;
 
+use std::ffi::c_void;
+use std::ptr;
+use std::sync::atomic::Ordering;
+use crate::native::uworld::UClass;
+
```
Comment (oberien) [inline rtil/src/native/mod.rs:39]: unused imports

**Review by oberien — CHANGES_REQUESTED** [review]

File: rtil/src/native/character.rs (hunk 2 above)
Comment (LukeSaward1) [reply rtil/src/native/character.rs:87]: Until the ordering is fixed, this will stay unresolved.

File: rtil/src/native/hud.rs [hunk 7]
```diff
@@ -13,7 +14,11 @@ use crate::statics::Static;
 
 static AMYHUD: Lazy<Static<usize>> = Lazy::new(Static::new);
 
-pub struct AMyHud;
+#[repr(C)]
+pub struct AMyHud {
+    #[cfg(windows)] _pad: [u8; 0x2b8],
+    pub b_showhud: u8,
```
Comment (oberien) [inline rtil/src/native/hud.rs:20]: This field should be called `bitfield`, as it stored more than just `bShowHUD`. It should have a comment with all bits' names.

This is also how it's done e.g. in `FActorSpawnParameters`:
```rust
struct FActorSpawnParameters {
    ...
    // bRemoteOwned, bNoFail, bDeferConstruction, bAllowDuringConstructionScript
    bitfield: u8,
    ...
}
```

**Review by oberien — CHANGES_REQUESTED** [review]

**Review by oberien — APPROVED** [review]
Thanks for the PR! :tada:
